tiny_cnn/core/backend_avx.h

```diff
@@ -0,0 +1,285 @@
+/*
+    Copyright (c) 2016, Taiga Nomi, Edgar Riba
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+    * Redistributions of source code must retain the above copyright
+    notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+    notice, this list of conditions and the following disclaimer in the
+    documentation and/or other materials provided with the distribution.
+    * Neither the name of the <organization> nor the
+    names of its contributors may be used to endorse or promote products
+    derived from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+    EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+    WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+    DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY
+    DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+    (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+    ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+    (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+    SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+#pragma once
+
+#include "tiny_cnn/core/backend.h"
+
+#include "tiny_cnn/core/kernels/tiny_conv2d_kernel.h"
+#include "tiny_cnn/core/kernels/tiny_conv2d_back_kernel.h"
+#include "tiny_cnn/core/kernels/tiny_deconv2d_kernel.h"
+#include "tiny_cnn/core/kernels/tiny_deconv2d_back_kernel.h"
+#include "tiny_cnn/core/kernels/tiny_maxpool_kernel.h"
+#include "tiny_cnn/core/kernels/tiny_fully_connected_kernel.h"
+
+#ifndef CNN_USE_AVX
+#error Advanced Vector Extensions required.
+#endif
```

Comment:
@edgarriba While ```#error``` directive is acceptable from the portability point of view, we can remove this because all lines including this file is already guarded by ```#ifdef / #endif```. I don't have a strong opinion but I prefer to remove these lines, if anything.